### ISSUE TITLE
Differing results when compared to MCS and rugarch packages

### ISSUE THREAD

**Issue by sjvrensburg** [nielsaka/modelconf#1]
This is probably not the right place to ask, but my question on R-SIG-Finance went unanswered. There are three packages that provide routines for determining the superior set of models (SSM) according to the model confidence set (MCS) procedure. However, all three give quite different answers even when using the same random seed. Do you have any idea why?

The code below, [copied from my gist](https://gist.github.com/sjvrensburg/57c325f62c52b9df56c82ca41f81de28), demonstrates the issue:
```R
require(MCS)
require(rugarch)
require(modelconf)
require(parallel)
require(dplyr)

cl <- makeForkCluster(nnodes = 8)   # Running Linux, so can fork.
data(Loss)        # Loss data from the MCS package
loss_mat <- as.matrix(Loss[, 1:10]) # mcsTest requires a "plain" matrix
model_names <- colnames(loss_mat)

# Parameters for functions...
alpha <- 0.2
B <- 50000        # Make this big, in order to reduce the variability.
block_len <- 33   # Block length calculated as in MCS package.

# Calculate MCS using various packages. Takes a while...
set.seed(123456)
rugarch_pkg <- mcsTest(losses = loss_mat, alpha = alpha, nboot = B,
                       nblock = block_len, boot = "block")
set.seed(123456)
mcs_pkg <- MCSprocedure(Loss = loss_mat, alpha = alpha, B = B,
                        statistic = 'TR', k = block_len, cl = cl)
set.seed(123456)
modelconf_pkg <- estMCS(loss = loss_mat, test = "t.range", B = B,
                        l = block_len)
stopCluster(cl)

# Extract and format results for comparison.
rugarch_results <- cbind(Model = model_names[c(rugarch_pkg$excludedR,
                                               rugarch_pkg$includedR)],
                         pval = rugarch_pkg$pvalsR)

mcs_results <- mcs_pkg@show %>%
  as.data.frame %>%
  cbind(Model = row.names(mcs_pkg@show), .) %>%
  select(Model, MCS_R) %>%
  arrange(MCS_R)

modelconf_results <- modelconf_pkg %>%
  as.data.frame %>%
  cbind(Model = row.names(modelconf_pkg), .) %>%
  select(Model, `MCS p-val`) %>%
  arrange(`MCS p-val`)

# Inspect results
#   rugarch: Only sGARCH-ghyp & sGARCH-sged in SSM
print(tail(rugarch_results, 5))
# Model                   
# [6,] "sGARCH-sstd"  "0.06714"
# [7,] "sGARCH-snorm" "0.06714"
# [8,] "sGARCH-jsu"   "0.06714"
# [9,] "sGARCH-ghyp"  "0.33244"
# [10,] "sGARCH-sged"  "1"    

#   MCS: only returns SSM, which includes 7 models.
print(tail(mcs_results, 5))
# 3  sGARCH-ghyp 0.82144
# 4   sGARCH-ged 0.96256
# 5   eGARCH-std 0.96704
# 6 sGARCH-snorm 0.99584
# 7  sGARCH-sged 1.00000

#   modelconf: SSM includes 7 models.
print(tail(modelconf_results, 5))
# Model MCS p-val
# 6   sGARCH-ghyp   0.70332
# 7    sGARCH-ged   0.87124
# 8  sGARCH-snorm   0.87124
# 9    eGARCH-std   0.87124
# 10  sGARCH-sged   1.00000
```
I do realise that the random seed probably doesn't work quite the same with parallelisation in `MCSprocedure`. Running it in serial doesn't, however, change the results. Also, differences in p-values are to be expected with bootstrapping. However, the p-values seem quite different and the `MCSprocedure` doesn't result in any "ties". Also, rugarch produces a much smaller SSM.

At some point, I would like to compare this with Hansen's [MULLCOM Ox package](https://9b31dea8-a-62cb3a1a-s-sites.googlegroups.com/site/peterreinhardhansen/research-papers/atestforsuperiorpredictiveability/MULCOM.zip?attachauth=ANoY7cr4-Tkbnv89TP1oYWipFDSRZ4N_1YC3g_xk5mqzqx8A-QZkmtOJw6g9HcFXkrvQKvX2XMhBP25aBOtv7DotMvnBxT0ekwREHf1xM2Wpva6bFYqKHzyJi0xHyhoEtwoZd98sndIXddeDxx1WSa4UHmGZyiAQQhjyrpcOBV59xUpDs7H2DdsU8eQVG3_TZ6SryUGaOlC01lRGcBpttUWScZFwyiYxdjQKXJze2cE22aDGRyHv5AdLcRnzrTXeAEBdQYWso0xVDtWtyz9K9jIqRuhNXVaIF4eCbsWFBZwRyd9MEjG6wDg%3D&attredirects=0), but I haven't quite figured out how to prepare the data for it. Have you been able to test it against MULLCOM? Unfortunately, the code is compiled as an Ox object. The code is therefore not viewable.

Obviously, you can't speak as to the quality of other people's code. But all of this has left me somewhat confused and unsure about the results I'm getting. So, I would very much appreciate any help or advice you might be have.

**Comment by Mofangjian** [comment]
I have the same problem. Have you solved it yet？
